src/active-response/firewalls/npf.c

```diff
@@ -0,0 +1,113 @@
+#define NPFCTL      "/sbin/npfctl"
+
+int main (int argc, char **argv) {
+    (void)argc;
+    char input[BUFFERSIZE];
+    char npf_active[COMMANDSIZE];
+    char npf_wazuh_ready[COMMANDSIZE];
+    char log_msg[LOGSIZE];
+    static char *srcip;
+    static char *action;
+    static char *iptables;
+    static cJSON *input_json = NULL;
+
+    write_debug_file(argv[0], "Starting");
+    // Reading input
+    memset(input, '\0', BUFFERSIZE);
+    if (fgets(input, BUFFERSIZE, stdin) == NULL) {
+        write_debug_file(argv[0], "Cannot read input from stdin");
+        return OS_INVALID;
+    }
+    write_debug_file(argv[0], input);
+
+    input_json = get_json_from_input(input);
+    if (!input_json) {
+        write_debug_file(argv[0], "Invalid input format");
+        return OS_INVALID;
+    }
+
+    action = get_command(input_json);
+    if (!action) {
+        write_debug_file(argv[0], "Cannot read 'command' from json");
+        cJSON_Delete(input_json);
+        return OS_INVALID;
+    }
+
+    if (strcmp("add", action) && strcmp("delete", action)) {
+        write_debug_file(argv[0], "Invalid value of 'command'");
+        cJSON_Delete(input_json);
+        return OS_INVALID;
+    }
+
+    // Get srcip
+    srcip = get_srcip_from_json(input_json);
+    if (!srcip) {
+        write_debug_file(argv[0], "Cannot read 'srcip' from data");
+        cJSON_Delete(input_json);
+        return OS_INVALID;
+    }
+
+    cJSON_Delete(input_json);
+
+    if (access(NPFCTL, F_OK) < 0) {
+        write_debug_file(argv[0], "The NPFCTL is not accessible");
+        return OS_INVALID;
+    }
+
+    memset(npf_active, '\0', COMMANDSIZE);
+    snprintf(npf_active, COMMANDSIZE -1,"%s show | grep 'filtering:' | grep -c active", NPFCTL);
+    if (system(npf_active) != 0) {
+        char log_msg[LOGSIZE] = "";
+        snprintf(log_msg, LOGSIZE -1, "Unable execute the command: '%s' ", npf_active);
+        write_debug_file(argv[0], log_msg);
+        return OS_INVALID;
+    }
+
+    memset(npf_wazuh_ready, '\0', COMMANDSIZE);
+    snprintf(npf_wazuh_ready, COMMANDSIZE -1,"%s show | grep -c 'table <wazuh_blacklist>'", NPFCTL);
+    if (system(npf_wazuh_ready) != 0) {
+        char log_msg[LOGSIZE] = "";
+        snprintf(log_msg, LOGSIZE -1, "Unable execute the command: '%s' ", npf_wazuh_ready);
+        write_debug_file(argv[0], log_msg);
+        return OS_INVALID;
+    }
+
+    char *exec_cmd[6];
+    if (!strcmp("add", action)) {
+        char *arg[6] = {NPFCTL, "table", "wazuh_blacklist", "add", srcip, NULL};
+        memcpy(exec_cmd, arg, sizeof(exec_cmd));
+
+    } else {
+        char *arg[6] = {NPFCTL, "table", "wazuh_blacklist", "del", srcip, NULL};
+        memcpy(exec_cmd, arg, sizeof(exec_cmd));
+
+    }
+
+    // Executing it
+    wfd_t *wfd = wpopenv(NPFCTL, exec_cmd, W_BIND_STDOUT);
+    if(!wfd) {
+        memset(log_msg, '\0', LOGSIZE);
+        snprintf(log_msg, LOGSIZE - 1, "Error executing %s : %s", NPFCTL, strerror(errno));
```

Comment:
Use this log:
`"Error executing '%s' : %s"`

Reply:
Done